src/levels/handler.rs

```diff
@@ -35,39 +46,187 @@ impl LevelHandler {
         }
     }
 
+    /// Replaces tables with given tables. This is done during loading.
+    pub fn init_tables(&mut self, tables: Vec<Table>) {
+        self.tables = tables;
+        self.total_size = 0;
+        for table in &self.tables {
+            self.total_size += table.size();
+        }
+
+        if self.level == 0 {
+            // Key range will overlap. Just sort by file id in ascending order
+            // because newer tables are at the end of level 0.
+            self.tables.sort_by_key(|x| x.id());
+        } else {
+            // Sort tables by keys.
+            self.tables
+                .sort_by(|x, y| COMPARATOR.compare_key(x.smallest(), y.smallest()));
+        }
+    }
+
+    pub fn delete_tables(&mut self, to_del: &[Table]) -> Result<()> {
+        let mut to_del_map = HashSet::new();
+
+        for table in to_del {
+            to_del_map.insert(table.id());
+        }
+
+        let mut new_tables = vec![];
+
+        for table in &self.tables {
+            if !to_del_map.contains(&table.id()) {
+                new_tables.push(table.clone());
```

Comment:
should use table instead of table.clone() if the for above does not use reference